plugins/typescript/src/generators/generateReactQueryFunctions.ts

```diff
@@ -155,7 +155,7 @@ export const generateReactQueryFunctions = async (
           Valid options: "useMutate", "useQuery"`);
         }
 
-        if (component === "useQuery") {
```

Comment:
I will be surprise than you do a mutation at the router level 🤔 I will keep this condition, you could also filter all the `useMutation` on this part to avoid trouble

Reply:
yes indeed you can exactly do an action at router level now. there is a `loader` and an `action` possibility. 
inside the actions you can mutate what u have. see:
https://reactrouter.com/en/main/route/action

Reply:
and even if i do not use the router actions i currently changed from the `useComponent()` to `useQuery()` or `useMutation()` inside of the components. like:

```JSX
// useQuery:
// const isLoggedIn = useGetUserLogin()
const isLoggedIn = useQuery(...getUserLoginQuery({}));

// useMutation:
// const loggoutUser = useUserLogout();
const loggoutUser = useMutation(...userLogoutQuery({}));

and one think i like here very much is that i see if it is an mutation or a get request.

Reply:
I really need to play with react-router to see how everything works together, but so far I don't see any usage of a mutationKey in their example.

You also need to invalidate the data manually:
https://reactrouter.com/en/main/guides/data-libs#invalidating-data-in-mutations

One thing that I don't like in your last example: it's very easy to use a mutation in a query (nothing will stop you to do it)

If you want an easy way to know what's a mutation or a query, you can try this -> https://github.com/fabien0102/openapi-codegen/issues/80#issuecomment-1229135698